scenarioo-client/test/protractorE2E/util/util.ts

```diff
@@ -0,0 +1,112 @@
+    await navigateToRoute('/');
+    await browser.executeScript(() => {
+        const injector = angular.element(document.body).injector();
+        const LocalStorageService: any = injector.get('LocalStorageService');
+        LocalStorageService.clearAll();
+        LocalStorageService.set('scenariooPreviouslyVisited', 'true');
+    });
+    const visited = await browser.executeScript(() => {
+        const injector = angular.element(document.body).injector();
+        const LocalStorageService: any = injector.get('LocalStorageService');
+        return LocalStorageService.get('scenariooPreviouslyVisited');
+    });
+    await expect(visited).toEqual('true');
+}
+
+/**
+ * Start Scenarioo as a user that has never visited it before (visited cookie will not be set).
+ */
+export async function startScenariooFirstTimeVisit() {
+    await clearLocalStorage();
+    await refreshBrowser(); // reload needed to restart without cookies.
+}
+
+/**
+ * Clear local storage to to simulate "first time user" for web tests to run with about dialog open.
+ *
+ * To really see the dialog, a restart of the application is needed after (e.g. by page refresh!),
+ * this method only clears the storage.
+ */
+export async function clearLocalStorage() {
+    await navigateToRoute('/');
+    await browser.executeScript(() => {
+        const injector = angular.element(document.body).injector();
+        const LocalStorageService: any = injector.get('LocalStorageService');
+        return LocalStorageService.clearAll();
+    });
+    const visited = await browser.executeScript(() => {
+        const injector = angular.element(document.body).injector();
+        const LocalStorageService: any = injector.get('LocalStorageService');
+        return LocalStorageService.get('scenariooPreviouslyVisited');
+    });
+    await expect(visited).toBe(null);
+}
+
+export async function assertRoute(route) {
+    const url = await browser.getCurrentUrl();
+    const currentUrlWithoutSearch = url.split('?')[0];
+    await expect(currentUrlWithoutSearch).toBe(browser.params.baseUrl + '/#' + route);
+}
+
+export async function assertPageIsDisplayed(path: string) {
+    return assertRoute(path);
+}
+
+export async function assertTextPresentInElement(e, expectedText) {
+    await expect(await e.getText()).toContain(expectedText);
+}
+
+export async function assertElementNotPresentInDom(elementBy) {
+    const elements = await browser.findElements(elementBy);
+    await expect(elements.length).toBe(0);
+}
+
+export async function clickBrowserBackButton() {
+    return browser.navigate().back();
+}
+
+export async function refreshBrowser() {
+    return browser.navigate().refresh();
+}
+
+export async function assertElementIsEnabled(elementId) {
+    return expect(element(by.id(elementId)).getAttribute('disabled')).toBeFalsy();
+}
+
+export async function assertElementIsDisabled(elementId) {
+    return expect(element(by.id(elementId)).getAttribute('disabled')).toBeTruthy();
```

Comment:
I know you did not write this - but a remark for future improvements - I think it would be better style to not pass ids or css classes to helper methods to identify elements, but use elements directly as params and pass them to such similar helper methods or even components. 

Advantage: these same functions can then be used no matter whether I used an id or a css selector or whatsoever to resolve the element. Also elements can even be easily reused in protractor, no problem at all.  

In general I think WebdriverJS elements are already a nice way of defining elements and we do not need to replace / wrap that by our own methods.

Reply:
Agree, but won't change as part of this PR as it's not in the scope of this change.

Reply:
sure